Content.Shared/Follower/FollowerSystem.cs

```diff
@@ -103,12 +152,15 @@ public void StartFollowingEntity(EntityUid follower, EntityUid entity)
             RemComp<FollowedComponent>(target);
         RemComp<FollowerComponent>(uid);
 
-        var xform = Transform(uid);
-        xform.AttachToGridOrMap();
-        if (xform.MapID == MapId.Nullspace)
+        if (deparent)
         {
-            Del(uid);
-            return;
+            var xform = Transform(uid);
+            xform.AttachToGridOrMap();
+            if (xform.MapID == MapId.Nullspace)
+            {
+                Del(uid);
+                return;
+            }
         }
```

Comment:
Does this cause issues on grid deletions / round restarts?

Reply:
StopFollowingEntity is called on grid deletions.

Reply:
I'm just going to remove it and let callers handle it as it stops the cleanup events being raised which will cause other issues if anyone relies on those events.